### PR TITLE
Test x86_64-pc-windows-gnu with Wine on CI

### PR DESCRIPTION
**Pull request by taiki-e**
This adds test for x86_64-pc-windows-gnu with Wine 7.13 on CI.

~~This also documents that Wine 7.13+ is supported.~~ EDIT: see https://github.com/tokio-rs/mio/pull/1637#issuecomment-1428318706

cc https://github.com/tokio-rs/mio/issues/1444

### CODE REVIEW

**Comment by taiki-e** [comment]
Hmm...

https://github.com/tokio-rs/mio/actions/runs/3583801507/jobs/6029672205

```
0380:fixme:dbghelp_dwarf:dwarf2_parse_udt_type Unhandled Tag type 0x33 at debug_info(abbrev:000000000056C6F0,symt:00000000003F53C4) in ctx(00000000003E5EF8,L"tcp_stream-4c63397ba5a78afa")

....

winedbg: Internal crash at 00000003AF6D8FE0
error: test failed, to rerun pass `--test tcp_stream`
```

I'll try more newer wine...

**Comment by taiki-e** [comment]
Added `#[cfg_attr(wine, ignore = "fails on Wine")]` to 5 tests that don't work on wine (https://github.com/tokio-rs/mio/pull/1637/commits/969624f821424a85d928a02f3faddba7ef2a4287).

The failures are all in the following form, and I'm not sure what is occurring.

```
wine: Unhandled page fault on read access to 0000000000000000 at address 00000003AF6D8FE0 (thread 0330), starting debugger...
0360:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
0360:err:winediag:nodrv_CreateWindow L"Make sure that your X server is running and that $DISPLAY is set correctly."
test shutdown_both ... Unhandled exception: page fault on read access to 0x0000000000000000 in 64-bit code (0x000003af6d8fe0).
winedbg: Internal crash at 00000003AF6D8FE0
```

File: README.md [hunk 1]
```diff
@@ -139,6 +139,7 @@ Currently supported platforms:
 * Windows
 * iOS
 * macOS
+* Wine (version 7.13+, see [issue #1444])
```
Comment (Thomasdezeeuw) [inline README.md:142]: Let's revert these changes as we still have a number of test failures.

Comment (taiki-e) [reply README.md]: Reverted doc changes and added note about cfg(wine) to CI config. (https://github.com/tokio-rs/mio/pull/1637/commits/16f6789b00edfb54ec215012be2ab0fdb86a6da5)

**Comment by Thomasdezeeuw** [comment]
@taiki-e what is the status of this? I see that some tests are still ignored (that are not ignored on Windows), so it's still not 100% supported?

**Comment by taiki-e** [comment]
> I see that some tests are still ignored (that are not ignored on Windows), so it's still not 100% supported?

Yeah, some tests are still failing (even with the latest wine), so they are ignored.

Ideally, I would like to fix the failure here, but I don't have the time or knowledge to investigate that at this time. (Perhaps I need to start by testing windows-gnu on CI with windows host to know if the failure is wine-specific or a mingw issue.)

**Comment by Thomasdezeeuw** [comment]
I'm going to close this due to age and because it's incomplete. @taiki-e thanks for the effort!

**Pull request closed without merging**